### PR TITLE
Fix for error when displaying IP allow lists containing custom IPs

### PR DESCRIPTION
**Pull request by mattops**

### CODE REVIEW

File: server/routes/components.ts [hunk 1]
```diff
@@ -117,17 +117,23 @@ export default function routes({ serviceCatalogueService, redisService }: Servic
 
     if (environments[0].ip_allow_list && environments[0].ip_allow_list_enabled) {
       const ipAllowListFiles = Object.keys(environments[0].ip_allow_list)
-      allowList.set('groups', [])
 
       ipAllowListFiles.forEach(fileName => {
         if (Object.keys(environments[0].ip_allow_list[fileName]).length > 0) {
```
Comment (psoleckimoj) [inline server/routes/components.ts:122]: Can probably ditch the `if` now as the `foreach` simply won't do anything if the length is 0. Or should behave that way

File: server/routes/components.ts [hunk 2]
```diff
@@ -117,17 +117,23 @@ export default function routes({ serviceCatalogueService, redisService }: Servic
...
+              const genericService = environments[0].ip_allow_list[fileName]['generic-service']
+              Object.keys(genericService).forEach(ipName => {
+                if (ipName !== 'groups') {
+                  allowList.set(ipName, genericService[ipName])
+                } else {
+                  allowList.set(ipName, Array.from([...new Set([...allowList.get(ipName), ...genericService[ipName]])]))
+                }
+              })
+            }
+            if (item !== 'generic-service') {
```
Comment (psoleckimoj) [inline server/routes/components.ts:135]: Just need an `else` rather than another `if`

Comment (mattops) [reply server/routes/components.ts:135]: not sure about that, what if we have both types?

Comment (mattops) [reply server/routes/components.ts]: ok I see

**Review by psoleckimoj — APPROVED** [review]
LGTM!

**Pull request merged**